pkg/engine/internal/scheduler/wire/wire.go

```diff
@@ -0,0 +1,57 @@
+// Package wire provides the wire protocol for how peers scheduler peers
+// communicate.
+package wire
+
+import (
+	"context"
+	"errors"
+	"net"
+)
+
+// ErrConnClosed indicates a closed connection between peers.
+var ErrConnClosed = errors.New("connection closed")
+
+// Listener waits for incoming connections from scheduler peers.
+type Listener interface {
+	// Accept waits for and returns the next connection to the listener. Accept
+	// returns an error if the context is canceled or if the listener is closed.
+	Accept(ctx context.Context) (Conn, error)
+
+	// Close closes the listener. Any blocked Accept operations will be
+	// unblocked and return errors.
+	Close() error
```

Comment:
`context.Context` argument would be beneficial here. E.g., http transport needs to gracefully shutdown an underlying http server which takes some time. A context would allow the caller to better control graceful shutdown (e.g., shutdown immediately and ignore active connections).